contracts/pallet-ibc/src/lib.rs

```diff
@@ -589,19 +590,20 @@ pub mod pallet {
 				.ok_or_else(|| Error::<T>::InvalidAssetId)?;
 
 			let account_id_32: AccountId32 = origin.clone().into();
-			let from = runtime_interface::account_id_to_ss58(account_id_32.into())
-				.and_then(|val| {
-					String::from_utf8(val).map_err(|_| SS58CodecError::InvalidAccountId)
-				})
-				.map_err(|_| Error::<T>::Utf8Error)?;
+			let from = runtime_interface::account_id_to_ss58(
+				account_id_32.into(),
+				<T as frame_system::Config>::SS58Prefix::get(),
+			)
+			.and_then(|val| String::from_utf8(val).map_err(|_| SS58CodecError::InvalidAccountId))
+			.map_err(|_| Error::<T>::Utf8Error)?;
 			let to = match params.to {
 				MultiAddress::Id(id) => {
+					// we convert id to hex string instead of ss58 because destination chain could
```

Comment:
Are you saying that the ss58 string may not be shared by the same user across both chains?
i.e. the '5abc...' account on chain 1 doesnt belong to the same user on chain 2?
or are you saying the 'from' user could be transferring funds to a different 'to' account?

If we convert one of these 'from' variables to hex then i think that should be consistent for 'to' also, across all the relevant ibc events

Reply:
Converted all to hex string now

Reply:
Shouldnt all the ibc events have matching logic like the "TransferInitiated" event for the from/to addresses being hexified? Or do these from/to hex strings simply get passed around and require no additional converting on events like "TransferFailed" "TokenReceived" etc?